grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -162,27 +188,62 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
 
                            final ArmeriaMessageDeframer deframer =
                                    new ArmeriaMessageDeframer(Integer.MAX_VALUE);
-                           msg.toHttpResponse().decode(deframer, ctx.alloc())
-                              .subscribe(singleSubscriber(msg, responseFuture), ctx.eventLoop());
+                           msg.toHttpResponse().decode(deframer, ctx.alloc(),
+                                                       byteBufConverter(ctx.alloc(), isGrpcWebText))
+                              .subscribe(singleSubscriber(ctx, msg, serializationFormat, responseFuture),
+                                         ctx.eventLoop());
                            return responseFuture;
                        }), ctx.eventLoop());
         }
 
         private static Subscriber<DeframedMessage> singleSubscriber(
-                AggregatedHttpResponse msg, CompletableFuture<HttpResponse> responseFuture) {
+                ClientRequestContext ctx, AggregatedHttpResponse msg, SerializationFormat serializationFormat,
+                CompletableFuture<HttpResponse> responseFuture) {
 
             return new Subscriber<DeframedMessage>() {
+                private HttpData content = HttpData.empty();
+                @Nullable
+                private HttpHeaders trailers;
+
                 @Override
                 public void onSubscribe(Subscription s) {
-                    s.request(1);
+                    // At least 2 requests are required for receiving trailers.
+                    s.request(2);
```

Comment:
This makes me wonder what would happen if we use this client for streaming responses. Should we handle such a case and raise an exception or warn?

Reply:
@trustin do you mean - complete exceptionally if subscriber receives a 2nd deframed message that's not a trailer? or something else?

Reply:
Yeah something like that. Sorry for a late reply.